src/Microsoft.TestPlatform.Utilities/CodeCoverageRunSettingsProcessor.cs

```diff
@@ -0,0 +1,484 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Xml;
+    using System.Xml.XPath;
+
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Represents the run settings processor for code coverage data collectors.
+    /// </summary>
+    public class CodeCoverageRunSettingsProcessor
+    {
+        #region Type Members
+        /// <summary>
+        /// Represents the exclusion type for code coverage run settings.
+        /// </summary>
+        private class Exclusion
+        {
+            /// <summary>
+            /// Represents the <see cref="XPathNavigator"/> style path of the exclusion type.
+            /// </summary>
+            private string path;
+
+            /// <summary>
+            /// Gets the path components for the current exclusion type.
+            /// </summary>
+            public IEnumerable<string> PathComponents { get; }
```

Comment:
similar like some other comment. IList is better in this case

Reply:
Done.